c3po/job/scheduler.py

```diff
@@ -1,13 +1,30 @@
+from logging import getLogger
+
 from apscheduler.schedulers.background import BackgroundScheduler
 
+from c3po.job.updater import update_posts_db
 from c3po.job.utils.lttkgp_db import first_time_init
+from c3po.utils.config import read_config
 
+LOG = getLogger(__name__)
 sched = BackgroundScheduler(daemon=True)
+PROCESS_POSTS_FREQUENCY = int(
+    read_config(section="scheduler")["PROCESS_POSTS_FREQUENCY"]
+)
+FB_FETCH_FREQUENCY = int(read_config(section="scheduler")["FB_FETCH_FREQUENCY"])
 
 
-def update_db():
-    first_time_init()
+def setup_scheduler(fetch_fb_posts=True, process_posts=True):
+    if fetch_fb_posts:
+        sched.add_job(update_posts_db, "interval", minutes=FB_FETCH_FREQUENCY)
+        LOG.info(
+            f"Add scheduler job to Fetch FB posts every {FB_FETCH_FREQUENCY} minutes"
+        )
 
+    if process_posts:
+        sched.add_job(first_time_init, "interval", minutes=PROCESS_POSTS_FREQUENCY)
+        LOG.info(
+            f"Add scheduler job to process Mongo posts every {PROCESS_POSTS_FREQUENCY} minutes"
+        )
```

Comment:
I like the idea of keeping them separate. Does this isolate failures too? What does failure look like? (Say, Graph API throws an exception)

Reply:
But with the current implementation, `first_time_init` fetches every post from mongo and tries to update our DB. But in reality, the number of posts that have actually changed would be 0-3 in a span of 15 minutes.

Would it be a better idea to stitch these 2 steps into 1 synchronous step? That way, we know exactly the diff of the posts that have changed and we can update only those posts on Postgres. What do you think?

Reply:
As discussed on chat, fixed this :+1: